addons/maintenance/models/maintenance.py

```diff
@@ -106,7 +106,11 @@ def _compute_maintenance_team_id(self):
     def _compute_maintenance_request(self):
         for record in self:
             maintenance_requests = record.maintenance_ids.filtered(lambda mr: mr.maintenance_type == 'corrective' and mr.stage_id.done)
-            record.mttr = len(maintenance_requests) and (sum(int((request.close_date - request.request_date).days) for request in maintenance_requests) / len(maintenance_requests)) or 0
+            mttr_days = 0
+            for request in maintenance_requests:
+                if request.stage_id.done and request.close_date:
+                    mttr_days += (request.close_date - request.request_date).days
+            record.mttr = len(maintenance_requests) and (mttr_days / len(maintenance_requests)) or 0
```

Comment:
Re-testing on `stage_id.done` is not useful (`maintenance_requests` already `filtered(...)`)

```suggestion
            record.mttr = len(maintenance_requests) and (sum(int(((request.close_date or request.request_date) - request.request_date).days) for request in maintenance_requests) / len(maintenance_requests)) or 0
```

As this occurs when migrating and #106166 solves in `write`, maybe an update pr is more appropriate ?

Reply:
OK let me try to do on upgrade . thanks for the review.